docs/content/migration.md

```diff
@@ -0,0 +1,575 @@
+| [layout][108] system                 | Covered in [separate section](#cdk-layout) of this guide  |
+
+### <a name="theming"></a> Theming
+
+AngularJS Material performs most theming at _run-time_. While this allows developers to configure
+themes dynamically in JavaScript, it also incurs substantial performance cost to generate
+CSS during application load.
+
+Angular Material performs theming at _compile-time_ with [Sass][170]. While this approach
+requires more up-front set-up, it completely removes all JavaScript execution cost associated
+with theming. The use of Sass also moves the theming system to a more well-supported and
+broadly-understood toolchain.
+
+#### Defining palettes
+
+AngularJS Material supports defining [custom palettes][151] or [extending existing palettes][158].
+
+In Angular Material, custom palette creation, including definition of hues and contrast colors, can
+be accomplished by copying one of the [provided palettes][167], modifying it, and then using it in
+your [custom theme creation][164].
+
+#### Defining themes
+
+AngularJS Material uses a provider, [`$mdThemingProvider`][144], to define the theme configuration
+used to generate styles during application loading. This provider also includes a default theme for
+cases when no custom palettes are provided.
+
+Angular Material has no default theme; a theme CSS file must _always_ be included. You can either
+use one of the [pre-built theme files][163], or define a [custom theme][164] with Sass mixins,
+including the output in your application just like any other CSS asset.
+
+##### Defining themes at run-time
+
+AngularJS Material uses a service, [`$mdTheming`][143], to define themes and generate CSS styles at
+run-time.
+
+Defining and generating theme CSS at run-time is not supported by Angular Material. However,
+loading additional theme CSS into your application (or modifying an existing theme's `link` node
+to [swap theme CSS files][181]) at run-time is supported.
+
+Your application can choose to generate additional themes at compile-time or build an API to
+dynamically generate theme CSS on-demand. In either case, it is possible to [lazy load][165] the
+theme CSS at run-time.
+
+#### Applying themes to your application
+
+AngularJS Material has a custom directive, [`md-theme`][155], to apply a theme to a specific
+component and its children.
+
+Rather than providing a special directive API, Angular Material's Sass-based theming system relies
+on standard Sass mixins and vanilla CSS selectors. For example, you can customize an individual
+button (or set of buttons) by defining and applying a class:
+
+```scss
+// Define a CSS class for a special button treatment.
+.super-special-button {
+  @include mat-button-theme($super-special-theme);
+}
+```
+```html
+<!-- Apply this treatment the same as you would with any other CSS class. -->
+<button mat-button class="super-special-button">Save</button>
```

Comment:
I realized after writing this that the class actually has to be on a parent element